src/storage/memory_store.ts

```diff
@@ -0,0 +1,697 @@
+  async addressCount(): Promise<number> {
+    return this.addresses.size;
+  }
+
+  /**
+   * Get the address info from its bip32 index.
+   * @param index bip32 address index to search for
+   * @async
+   * @returns {Promise<IAddressInfo | null>} The address info or null if not in storage
+   */
+  async getAddressAtIndex(index: number): Promise<IAddressInfo | null> {
+    const addr = this.addressIndexes.get(index);
+    if (addr === undefined) {
+      // We do not have this index loaded on storage, it should be generated instead
+      return null;
+    }
+    return this.addresses.get(addr) as IAddressInfo;
+  }
+
+  /**
+   * Save the address in storage
+   * @param {IAddressInfo} info Info on address to save
+   * @async
+   * @returns {Promise<void>}
+   */
+  async saveAddress(info: IAddressInfo): Promise<void> {
+    if (!info.base58) {
+      throw new Error('Invalid address');
+    }
+    if (this.addresses.has(info.base58)) {
+      throw new Error('Already have this address');
+    }
+
+    // Saving address info
+    this.addresses.set(info.base58, info);
+    this.addressIndexes.set(info.bip32AddressIndex, info.base58);
+
+    if (this.walletData.currentAddressIndex === -1) {
+      this.walletData.currentAddressIndex = info.bip32AddressIndex;
+    }
+
+    if (info.bip32AddressIndex > this.walletData.lastLoadedAddressIndex) {
+      this.walletData.lastLoadedAddressIndex = info.bip32AddressIndex;
+    }
+  }
+
+  /**
+   * Check that an address is in our storage.
+   * @param {string} base58 Address to check.
+   * @async
+   * @returns A promise that resolves to wheather the address is saved in storage or no.
+   */
+  async addressExists(base58: string): Promise<boolean> {
+    return this.addresses.has(base58);
+  }
+
+  /**
+   * Get the current address.
+   *
+   * @param {boolean | undefined} markAsUsed If we should set the next address as current
+   * @async
+   * @returns {Promise<string>} The address in base58 format
+   */
+  async getCurrentAddress(markAsUsed?: boolean): Promise<string> {
+    const addressInfo = await this.getAddressAtIndex(this.walletData.currentAddressIndex);
+    if (!addressInfo) {
+      throw new Error('Current address is not loaded');
+    }
+
+    if (markAsUsed) {
+      // Will move the address index only if we have not reached the gap limit
+      this.walletData.currentAddressIndex = Math.min(this.walletData.lastLoadedAddressIndex, this.walletData.currentAddressIndex + 1);
+    }
+    return addressInfo.base58;
+  }
+
+  /**
+   * Set the value of the current address index.
+   * @param {number} index The index to set
+   */
+  async setCurrentAddressIndex(index: number): Promise<void> {
+    this.walletData.currentAddressIndex = index;
+  }
+
+  /**
+   * Edit address metadata.
+   *
+   * @param {string} base58 The address in base58 format
+   * @param {IAddressMetadata} meta The metadata to save
+   */
+  async editAddress(base58: string, meta: IAddressMetadata): Promise<void> {
```

Comment:
I think `editAddressMeta` would be better. What do you think?

Reply:
I also think its better since it actually only affects the address meta.